YouTube.js

```diff
@@ -90,15 +90,14 @@ function doWeb(doc, url) {
 function scrape(doc, url) {
 	var item = new Zotero.Item("videoRecording");
 	item.title = text(doc, '#info-contents h1.title');
-	item.url = url;
+	item.url = ZU.xpathText(doc, '//link[@rel="canonical"]/@href') || url;
 	item.runningTime = text(doc, '#movie_player .ytp-time-duration');
 	
-	item.date = text(doc, '#info-text #date');
+	item.date = ZU.xpathText(doc, '//meta[@itemprop="datePublished"]/@content');
```

Comment:
And here:

```suggestion
	item.date = attr(doc, 'meta[itemprop="datePublished"]', 'content');
```

CSS selectors are more widely understood than XPath and lack its odd edge cases (namespaces in XHTML/XML documents...).